Update black and lint all python files

paulbarton90 opened the pull request:
### Update black and lint all python files

### Linked issues

N/A

### Summarize your change.

Updates black version used in the pre-commit hook from 22.8.0 to 24.8.0, and lints all python files via `pre-commit run -a black`.

### Describe the reason for the change.

To update black, and remove any linting issues in current files.

### Describe what you have tested and on which operating system.

Building in Centos 7

### Add a list of changes, and note any that might need special attention during the review.

* In going through the changes, in `src/plugins/rv-packages/otio_reader/otio_writer.py` on line 221, the `kwargs["pre_item"]` is passed in as a tuple. Is this meant to be a tuple, as the name is singluar.
* Please verify that the build of the documentation works as expected, since black has changed the quoting from `'` to `"`.

geffrak reviewed (changes requested):
First, thanks for updating black :) 

This PR is changing too many files. The only change I accept is https://github.com/AcademySoftwareFoundation/OpenRV/pull/534/commits/b1ee6dec156096abd385898183830ece782b26ab. Everything else needs to be reverted. I would prefer to have people modifying the files they touch with the pre-commit hooks rather than having one big commit like this one because 

1) It messes with the git history
2) There were cases of regressions introduced by supposedly safe linting and/or formatting code.

Unfortunately, our code coverage is far from enough to accept this pull request without an extensive pre merge test.

paulbarton90 commented:
> This PR is changing too many files. The only change I accept is [b1ee6de](https://github.com/AcademySoftwareFoundation/OpenRV/commit/b1ee6dec156096abd385898183830ece782b26ab). Everything else needs to be reverted. I would prefer to have people modifying the files they touch with the pre-commit hooks rather than having one big commit like this one because
> 
> 1. It messes with the git history
> 2. There were cases of regressions introduced by supposedly safe linting and/or formatting code.

No worries, I have removed the extra commits not related to updating black, and updated the title and description for the MR.

bernie-laberge reviewed (approved):
Hello @paulbarton90,
Thank you for your contribution !
